### DISCUSSION TITLE
manifold.count always zero when colliding with kinematic bodies?

### DISCUSSION THREAD

**Discussion by jensblomqvist** [bepu/bepuphysics2#210] [General]
I am adding kinematic bodies to my game engine and I noticed that ConfigureContactManifold is still called but manifold.count is always zero? My test involves colliding a dynamic with a kinematic body.

**Comment by RossNordby** [comment]
I wouldn't expect a dynamic-kinematic to behave differently than a dynamic-dynamic pair with respect to reporting contacts. It *is* expected that `ConfigureContactManifold` will sometimes be called with zero contacts; if bounding boxes overlap (and the callbacks permit testing), but the shapes are not within the pair's speculative margin, no contacts will be generated.

Does one of the bodies have a `Mesh` or `Triangle` shape? Those are one sided, so collisions on the backface don't generate contacts. Related, are you using 2.4, or an older version? 2.3 had some bugs around triangle contact generation that were fixed in 2.4 (and recently 2.3.1). 

If it doesn't appear to be related to any of the above, I'll probably need a repro in the demos to figure it out.

**Reply by jensblomqvist** [reply]
Thank you for the quick response!

In my simulation the kinematic bodies can be either spheres or boxes and the dynamics are compounds of spheres and boxes. This is all a bit strange beacuse I can't reproduce the problem anymore. At one point I couldn't get one kinematic to report any contact points, then it was random and now it works as expected. Maybe I found one very special case or I did something wrong. I will keep an eye on this one.
